### ISSUE TITLE
Removing stale IX entries

### ISSUE THREAD

**Issue by arnoldnipper** [peeringdb/peeringdb#299]
The AC is currently scutinizing all `ix` records to add missing `ixpfx` entries. In most cases this can easily be done by looking at exitisting `netixlan` records for the given `ix`.

We futhermore propose to **remove** `ix` records which then

 * do not have any `ixpfx` records

 * do not have any `netixlan` records

 * are older then x days (x = 90 seems reasonable)

**Comment by mahtin** [comment]
So I have a few thoughts.

1) The word “remove” is very harsh word in this context and quite unlike other actions within the PeeringDB world. I prefer at-least a two-step process. Warning message (including on UI plus onwards to the listed contacts for an IX) at N days. Removal (or setting invalid status) after M days.

2) 90 days is very strict. RIRs ask for records to be checked/updated once per year. That seems to be an acceptable standard. A recurring yearly check is only needed if no update has been performed (update date is an important field).

3) IXs are NOT (repeat NOT) responsible or capable-of adding member records (hence netixlan records). If an IX doesn’t have any members listed; sadly that’s not the IXs fault. (This pains me to say this).

4) I’ve come across quite a few IXs without ixpfx records. I email the IXs directly (esp’ if we are trying to connect). I believe PeeringDB should start being more proactive in areas like this. For example, if an IX record is created, then check N days later for an ixpfx record. If none found, then send email requesting it be added. QED.

So all this revolves around knowing when records are updated. I’m basically introducing the concept of a “freshness-factor” to this discussion. That can be used within a UI to show an IX’s state (green, orange, red - for freshness). It can trigger emails. It helps improve PeeringDB.

Summary: I’m against removing anything. I’m pro freshness indicators and proactive email notices. 

Martin

PS: I know the original PDB (1.0) never emailed anything. We are past that taboo.

**Comment by arnoldnipper** [comment]
@mahtin 

ad 1. we remove - or to be more precise flag a record "deleted - all the time within PeeringDB. This happens to net, ic and fac records. User records are even hard deleted.

ad 2. I guess you read *AND* as *OR* ... so please reread if it makes more sense then

ad 3. Again it's *AND* and not *OR*. If an IX has an `ixpfx` all is fine and it will not be removed

ad 4. See #291 i.e. new IX have to have LAN information aka `ixpfx` 

The purpose is to really get rid of IXes which do not exist

**Comment by arnoldnipper** [comment]
There are currently 31 IXes (out of currently 641) which don't have no ixpfx record and hence also no netixlan record associated to it). For some we even don't have tech_email info or policy_email.

I would like to propose 
- remove entries w/o email information and older than 180 days immediately
- contact all entries with email information twice (wait 4 weeks in between). Remove if no answer
[ixes_with_neither_ixpfx_nor_netixlan_information.txt](https://github.com/peeringdb/peeringdb/files/1822013/ixes_with_neither_ixpfx_nor_netixlan_information.txt)

**Comment by mahtin** [comment]
Arnold - Can you post the list of 31 IXs? Maybe someone has contacts they could add into the db.

**Comment by arnoldnipper** [comment]
@mahtin ... was attached to the ticket

**Comment by aftabsiddiqui** [comment]
Message has been sent to all IXPs in the APAC region. Hopefully, they will do the right thing :)

**Comment by grizz** [comment]
@arnoldnipper I agree with this, I can't tell if you're requesting a software feature or just having the AC do a 1 time delete? Moving to review pending further discussion.

Seems like this should maybe go into some documentation somewhere as well.

**Comment by arnoldnipper** [comment]
@mahtin 

> I’ve come across quite a few IXs without ixpfx records. I email the IXs directly (esp’ if we are trying to connect). I believe PeeringDB should start being more proactive in areas like this. For example, if an IX record is created, then check N days later for an ixpfx record. If none found, then send email requesting it be added. QED.

Meanwhile request for new IXes have to add `ixlan` and ìxpfx` info

**Comment by arnoldnipper** [comment]
@grizz , AC is happy to simply soft delete those IXes

**Comment by arnoldnipper** [comment]
AC has sent out a reminder to those IXes we have an email address from to add LAN information until 2018-12-31 23:59 UTC. All IXes missing LAN information will be removed starting 2019-01-01 00:00 UTC.

**Comment by grizz** [comment]
Awesome, is there a list available of which are still missing it?

**Comment by arnoldnipper** [comment]
attached
[201811290248-ixes-wo-ixpfx.txt](https://github.com/peeringdb/peeringdb/files/2627399/201811290248-ixes-wo-ixpfx.txt)

**Comment by arnoldnipper** [comment]
> AC has sent out a reminder to those IXes we have an email address from to add LAN information until 2018-12-31 23:59 UTC. All IXes missing LAN information will be removed starting 2019-01-01 00:00 UTC.

Did this again yesterday

**Comment by arnoldnipper** [comment]
> Seems like this should maybe go into some documentation somewhere as well.

Opened #409 to document
